### PR TITLE
Monorepo dependency update

### PR DESCRIPTION
**Pull request by ozgunozerk**
This PR consists of:
- updating monorepo and substrate dependencies
- further refactoring main, better structured code
- `tracing` instead of `logging`
- fix a small issue, where we display old farmed blocks even if they are farmed under a different reward address (now only shows the ones farmed under the current reward address)

Don't go commit by commit, reviewing altogether is fine.

### CODE REVIEW

**Comment by ozgunozerk** [comment]
Will merge when we update the chain spec (new one with new bootnodes)

File: src-tauri/Cargo.toml [hunk 1]
```diff
@@ -1,39 +1,59 @@
...
+version = "0.5.3"
 description = "Subspace desktop"
 authors = ["Subspace Labs <https://subspace.network>"]
 license = "Apache-2.0"
 repository = "https://github.com/subspace/subspace-desktop"
 edition = "2021"
 
 [build-dependencies]
-substrate-build-script-utils = { version = "3.0.0", git = "https://github.com/paritytech/substrate", rev = "5d3e7c4ee9f5c8e370022d5f3fa9723185710831" }
+substrate-build-script-utils = { version = "3.0.0", git = "https://github.com/paritytech/substrate", rev = "24bea4c3cba7479e2cf2976a21e8111dfda6b1cd" }
```
Comment (nazar-pc) [inline src-tauri/Cargo.toml:11]: We switched to our fork in https://github.com/subspace/subspace/pull/533 (same commit though) to make builds faster and use less disk space

File: src-tauri/src/node.rs [hunk 2]
```diff
@@ -181,7 +186,9 @@ fn create_configuration<CS: ChainSpec + 'static>(
             // TODO: Change to constrained eventually (need DSN for this)
             state_pruning: Some(PruningMode::keep_blocks(1024)),
             keep_blocks: KeepBlocks::Some(1024),
-            wasm_method: WasmExecutionMethod::Compiled,
+            wasm_method: WasmExecutionMethod::Compiled {
+                instantiation_strategy: WasmtimeInstantiationStrategy::PoolingCopyOnWrite,
+            },
             wasm_runtime_overrides: None,
             #[cfg(not(target_os = "windows"))]
```
Comment (nazar-pc) [inline src-tauri/src/node.rs:193]: This time Windows should be the same as other platforms, performance issue was fixed

File: src/components/farmedList.vue [hunk 3]
```diff
@@ -84,7 +84,10 @@ export default defineComponent({
   },
   computed: {
     farmedBlocksList(): FarmedBlock[] {
-      return this.$client?.data?.farming.farmed || []
+      const result = this.$client.data?.farming.farmed.filter((block) => {
+        return block.rewardAddr == this.rewardAddress
```
Comment (nazar-pc) [inline src/components/farmedList.vue:88]: @isSerge looks like this update doesn't know anything about votes, right? That is where the most of rewards will come from though. It is an unsigned extrinsic, see https://github.com/subspace/subspace/pull/485

File: src-tauri/src/utils.rs [hunk 4]
```diff
@@ -0,0 +1,85 @@
+use log::{debug, error, info};
+use serde::Serialize;
+use std::io::Write;
+use std::path::PathBuf;
+use tauri::{
+    api::{self},
```
Comment (nazar-pc) [inline src-tauri/src/utils.rs:6]: :shrug: 
```suggestion
    api,
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "subspace-desktop",
-  "version": "0.5.2",
+  "version": "0.5.3",
```
Comment (nazar-pc) [inline package.json:3]: Shouldn't it be `0.6.0` though?

File: src/components/farmedList.vue (hunk 3 above)
Comment (isSerge) [reply src/components/farmedList.vue:88]: yes, we will have a discussion on how do we display rewards today

File: src-tauri/Cargo.toml (hunk 1 above)
Comment (ozgunozerk) [reply src-tauri/Cargo.toml:11]: this is wonderful 🥳

**Review by nazar-pc — APPROVED** [review]

**Pull request merged**